### PR TITLE
Unify build script

### PR DESCRIPTION
**Pull request by SinTh0r4s**
~~DO NOT MERGE YET!~~

~~Draft to test unified build script. Works flawlessly in release, but does not run in dev. I assume it never did? Do you know where the problem comes from?~~ Fixed. Works fine in dev as well

### CODE REVIEW

File: repositories.gradle [hunk 1]
```diff
@@ -0,0 +1,19 @@
+repositories {
+    maven {
+        name = "gt"
+        url = "https://gregtech.overminddl1.com/"
+    }
+    maven {
+        // IC2
+        name = "ic2"
+        url = "http://maven.ic2.player.to/"
```
Comment (glowredman) [inline repositories.gradle:9]: You may use `https` here

Comment (SinTh0r4s) [reply repositories.gradle]: Thanks! Fixed :-)

**Comment by SinTh0r4s** [comment]
Does anybody have a clue why the CI run fails? Maybe it is a difference between `setupDecompWorkspace` and `setupCIWorkspace`?

File: dependencies.gradle [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    // GT5U Dependencies
+    compile("net.industrial-craft:industrialcraft-2:2.2.828-experimental:dev")
+    compile("com.github.GTNewHorizons:StructureLib:1.0.6:deobf")
+    compile("codechicken:CodeChickenLib:1.7.10-1.1.3.140:dev")
+    compile("codechicken:CodeChickenCore:1.7.10-1.0.7.47:dev")
+    compile("codechicken:NotEnoughItems:1.7.10-1.0.5.120:dev")
+
+    compileOnly("applecore:AppleCore:1.7.10-1.3.9:deobf")
+    compileOnly("thaumcraft:Thaumcraft:1.7.10-4.2.3.5:dev")
+    compileOnly files("dependencies/HungerOverhaul-1.7.10-1.0.2.jar")
```
Comment (glowredman) [inline dependencies.gradle:16]: No need to include a jar file, HungerOverhaul unless you need version 1.0.2:
```groovy
maven {
  name = "hungeroverhaul"
  url = "https://dvs1.progwml6.com/files/maven/"
}
```
and
```groovy
compileOnly("iguanaman:HungerOverhaul:1.7.10-1.0.0.jenkins104:deobf")
```

File: dependencies.gradle [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    compileOnly files("dependencies/CoFHCore-[1.7.10]3.1.4-329-dev.jar")
+    compileOnly files("dependencies/Railcraft_1.7.10-9.12.2.0-dev.jar")
```
Comment (glowredman) [inline dependencies.gradle]:
```groovy
compileOnly "mods.railcraft:Railcraft_1.7.10:9.12.3.0:dev"
```
(hosted on https://gregtech.overminddl1.com/)

File: build.gradle [hunk 4]
```diff
@@ -1,190 +1,376 @@
...
+import com.github.jengelman.gradle.plugins.shadow.tasks.ConfigureShadowRelocation
+import java.util.concurrent.TimeUnit
+
 buildscript {
     repositories {
-        mavenCentral()
         maven {
-            name = "gt"
-            url = "https://gregtech.overminddl1.com/"
+            name = "jitpack"
```
Comment (mitchej123) [inline build.gradle:15]: Do these get tried in order until it finds a match?  If so, Jitpack should be last.

**Comment by SinTh0r4s** [comment]
Mergable, but will get some minor updates down the line

**Review by mitchej123 — APPROVED** [review]

**Pull request merged**